WooCommerce/src/main/kotlin/com/woocommerce/android/ui/payments/methodselection/SelectPaymentMethodViewModel.kt

```diff
@@ -474,6 +486,12 @@ class SelectPaymentMethodViewModel @Inject constructor(
         return currencyFormatter.formatCurrency(total, currencyCode)
     }
 
+    private fun isWooPOSPaymentFlow() = if (navArgs.cardReaderFlowParam is Payment) {
+        (navArgs.cardReaderFlowParam as Payment).paymentType != WOO_POS
+    } else {
+        true
+    }
```

Comment:
Can we simplify the code like this?
```suggestion
    private fun isWooPOSPaymentFlow() = with(navArgs.cardReaderFlowParam) {
        this is Payment && paymentType == WOO_POS
    }
```

... and revert the `displayUi` value.

Reply:
Also – I'm wondering if maybe we can already loosen the type restriction to `PaymentOrRefund` 🤔

Reply:
Done here as well b736ee9c94cfa714ba11bf8cc6b28cce450b7782

> Also – I'm wondering if maybe we can already loosen the type restriction to PaymentOrRefund 🤔

Could you please elaborate on this one?

Reply:
Let's park this for now. I was thinking about handling refunds, but it will be a different flow.